src/main/java/loci/common/Location.java

```diff
@@ -145,7 +129,41 @@ public Location(File file) {
    * @param child the relative path name
    */
   public Location(String parent, String child) {
-    this(parent + File.separator + child);
+    LOGGER.trace("Location({}, {})", parent, child);
+
+    String mapped = null;
+    String pathname = null;
+
+    // First handle possible URIs
+    if (child != null && child.contains("://")) {
+      // Avoid expensive exception handling in case when path is
+      // obviously not an URL
+      try {
+        mapped = getMappedId(child);
+        pathname = child;
+        uri = new URI(mapped);
+        isURL = true;
+        url = uri.toURL();
+      }
+      catch (URISyntaxException | MalformedURLException e) {
+        // TODO: this should possibly throw
+        // possibly leaves url null
+      }
+    }
+
+    // If not a URI, then deal with relative vs. absolute paths
+    if (pathname == null) {
+      if (parent != null) {
```

Comment:
:+1: for handling this case. Note the fix had already been part of https://github.com/ome/ome-common-java/pull/12 but the change had been reverted given the breaking nature of it. As we might be moving towards `ome-common:6.0.0`, we can certainly re-open the change as a follow-up of the S3 work.